crates/sui-storage/src/write_path_pending_tx_log.rs

```diff
@@ -48,15 +44,20 @@ impl WritePathPendingTransactionLog {
         tx: &VerifiedTransaction,
     ) -> SuiResult<IsFirstRecord> {
         let tx_digest = tx.digest();
-        let _guard = self.mutex_table.acquire_lock(*tx_digest).await;
-        if self.pending_transactions.logs.contains_key(tx_digest)? {
-            Ok(false)
-        } else {
-            self.pending_transactions
-                .logs
-                .insert(tx_digest, tx.serializable_ref())?;
-            Ok(true)
+        let transaction = self.pending_transactions.logs.transaction()?;
+        if transaction
+            .get_for_update(&self.pending_transactions.logs, tx_digest)?
```

Comment:
I don't think we need a `get_for_update()` but a regular `get()` should do the job because we are anyways inserting the key in the transaction (which does write-write conflict detection by default)